src/components/EditableText.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { createRef, Fragment, Component } from "react";
+
+type InputState = {
+  initialValue: string;
+  value: string;
+  edit: boolean;
+};
+
+type Props = {
+  value: string;
+  updateName: (name: string) => void;
+};
+
+export default class EditableInput extends Component<Props, InputState> {
+  constructor(props: Props) {
+    super(props);
+    const { value } = props;
+
+    this.state = {
+      initialValue: value,
```

Comment:
Shall `initialValue` be updated when component receives new `value` from props?

Reply:
So initial value keeps track of the original name to fallback to, in the case that the user clears the input.
I figured this is a better thing to do than alert('field can't be empty')
Perhaps there is a better alternative? Happy to change

Reply:
I think you can always use `this.props.value` instead of the `this.state.initialValue` :) Can `initialValue` be removed from the state?

Reply:
oh yea totally, figured. will push the latest changes now. thanks for reviewing!